### ISSUE TITLE
"load iframe after accepting" usable with textmedia?

### ISSUE THREAD

**Issue by peerlight** [DirkPersky/typo3-dp_cookieconsent#68]
Is there any way to use "load iframe after accepting" with videos in CE textmedia?

**Comment by DirkPersky** [comment]
Yeah it is possible. You need to change the Fluid Templates of that element.

But actually i don't have an example for you ;((

**Comment by peerlight** [comment]
Hi, first attempt was using TS in lib.contentElement.settings.media.additionalConfig but you are right, FLUID has to be changed

Had a look at fluid_styled_content/Resources/Private/Partials/Media/Rendering/Video.html and also  bootstrap_package/Resources/Private/Partials/ContentElements/Media/Rendering/Video.html 

They both use <f:media> and I have no idea how to iframe rendering  via <f:media>

Do you recommend coding iframe manually? 

mindshape_cookie_consent uses a viewhelpfer for this
https://github.com/mindshape-GmbH/mindshape_cookie_consent/blob/master/Classes/ViewHelpers/MediaViewHelper.php

**Comment by DirkPersky** [comment]
If you use the [VHS](https://extensions.typo3.org/extension/vhs) Plugin.

Than u can modify the `Ressources/Private/Partials/Media/Type/Video.html` with the contains condition.
- https://viewhelpers.fluidtypo3.org/fluidtypo3/vhs/2.1.2/Condition/String/Contains.html
- https://stackoverflow.com/questions/31406639/fluid-condition-with-containsviewhelper#answer-31409560

**Comment by peerlight** [comment]
What would your condition check?

I was able to read {file.properties.extension} and check for "youtube" without using vhs and then create the iframe with a manually composed video URL.

Unfortunally using YoutubeRenderer is not possible - creates iframe with src= what is not wanted ...

**Comment by mxwssl** [comment]
> What would your condition check?
> 
> I was able to read {file.properties.extension} and check for "youtube" without using vhs and then create the iframe with a manually composed video URL.
> 
> Unfortunally using YoutubeRenderer is not possible - creates iframe with src= what is not wanted ...

You're able to override the YouTubeRenderer Class in your Extension's ext_localconf.php. Then copy the file from TYPO3s source and change the HTML in your own.

```
// Register your own online custom helper class
$GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['onlineMediaHelpers']['youtube'] = \Your\Extension\Resource\OnlineMedia\Helpers\YouTubeHelper::class;

$rendererRegistry = \TYPO3\CMS\Core\Resource\Rendering\RendererRegistry::getInstance();
$rendererRegistry->registerRendererClass(\Your\Extension\Resource\Rendering\YouTubeRenderer::class); 
```
